src/main/groovy/nebula/plugin/publishing/verification/VerificationReportTask.groovy

```diff
@@ -13,17 +13,16 @@ abstract class VerificationReportTask extends DefaultTask {
 
     protected VerificationReportGenerator verificationReportGenerator = new VerificationReportGenerator()
 
-    @Internal
-    abstract Property<PublishVerificationPlugin.VerificationViolationsCollectorHolderExtension> getVerificationViolationsCollectorHolderExtension()
-
     @Input
     abstract Property<String> getTargetStatus()
 
+    @Internal
+    abstract Property<VerificationViolationsCollectorService> getVerificationViolationsCollectorService()
+
     @TaskAction
     void reportViolatingDependencies() {
-        if (project.rootProject == project) {
-            reportErrors(verificationViolationsCollectorHolderExtension.get().collector)
-        }
+        reportErrors(verificationViolationsCollectorService.get().collector)
```

Comment:
Removed the need for checking if the project is a rootProject. We register the task only for the `rootProject` so we should be good with the assumption